ethpm/backends/base.py

```diff
@@ -0,0 +1,34 @@
+from abc import (
+    ABC,
+    abstractmethod,
+)
+
+
+class BaseURIBackend(ABC):
+    """
+    Generic backend that all URI backends are subclassed from.
+
+    All subclasses must implement:
+    can_handle_uri, fetch_uri_contents, base_uri
+    """
+    @abstractmethod
+    def can_handle_uri(self, uri: str) -> bool:
+        """
+        Return a bool indicating whether this backend class can handle the given URI.
+        """
+        raise NotImplementedError("Must be implemented by subclass.")
+
+    @abstractmethod
+    def fetch_uri_contents(self, uri: str) -> bytes:
+        """
+        Fetch the contents stored at a URI.
+        """
+        raise NotImplementedError("Must be implemented by subclass.")
+
+    @property
+    @abstractmethod
+    def base_uri(self) -> str:
```

Comment:
Does this make sense as a required base method/property?  Seems like IPFS backend wouldn't use this. since they are just a scheme and then the hash `ipfs://Qm...`

This might belong in a second *base* class for `HTTP` based backends since they are likely to share similar logic.

Reply:
Maybe not. In the local IPFS node backend, I used this property to return `host:port` where the IPFS daemon is running. But I could see that maybe being confusing/ incorrectly named? And moving this property up a level to only `HTTP` based backends

Reply:
I think I agree with @pipermerriam here.  Seems like `base_uri` should be in a subclass for HTTP backends.